src/Plutip/Server.purs

```diff
@@ -162,8 +189,69 @@ startOgmios cfg params = liftEffect $ spawn "ogmios" ogmiosArgs
     , params.nodeConfigPath
     ]
 
-stopOgmios :: ChildProcess -> Aff Unit
-stopOgmios = liftEffect <<< kill SIGINT
+stopChildProcess :: ChildProcess -> Aff Unit
+stopChildProcess = liftEffect <<< kill SIGINT
+
+startPlutipServer :: PlutipConfig -> Aff ChildProcess
+startPlutipServer cfg = do
+  p <- liftEffect $ spawn "plutip-server" [ "-p", UInt.toString cfg.port ]
+    defaultSpawnOptions { detached = true }
+  void
+    $ recovering defaultRetryPolicy
+        ([ \_ _ -> pure true ])
+    $ const
+    $ stopPlutipCluster cfg
+  pure p
+
+startPostgresServer
+  :: PostgresConfig -> ClusterStartupParameters -> Aff ChildProcess
+startPostgresServer pgConfig _ = do
+  tmpDir <- liftEffect tmpdir
+  randomStr <- liftEffect $ uniqueId ""
+  let
+    workingDir = tmpDir <> "/" <> randomStr
+    databaseDir = workingDir <> "/postgres/data"
+  liftEffect $ void $ execSync ("initdb " <> databaseDir) defaultExecSyncOptions
+  pgChildProcess <- liftEffect $ spawn "postgres"
+    [ "-D"
+    , databaseDir
+    , "-p"
+    , UInt.toString pgConfig.port
+    , "-h"
+    , pgConfig.host
+    , "-k"
+    , workingDir <> "/postgres"
+    ]
+    defaultSpawnOptions
+  void $ recovering defaultRetryPolicy ([ \_ _ -> pure true ])
+    $ const
+    $ liftEffect
+    $ execSync
+        ( "psql -h " <> pgConfig.host <> " -p " <> UInt.toString pgConfig.port
+            <> " -d postgres"
+        )
+        defaultExecSyncOptions
+  liftEffect $ log =<< Bf.toString Encoding.UTF8 =<< execSync
```

Comment:
Let's use async versions of NodeJS API calls where possible. We should not block the interpreter if we can, since useful work can happen in background (query dispatching, etc)

Reply:
I understand your concerns, but in this case we really need to execute the postgres related commands synchronized. I dont want the creation of a user to occur before `postgres-server` is executed. And that is the case for all the calls of execSync.